x-pack/plugins/observability_ai_assistant/public/analytics/user_interaction.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RootSchema } from '@kbn/analytics-client';
+import type { Message } from '../../common';
+
+export const USER_SENT_PROMPT = 'observability_ai_assistant_chat_user_sent_prompt' as const;
+
+export interface UserInteractionTelemetryEvent {
+  eventType: typeof USER_SENT_PROMPT;
+  schema: RootSchema<Message>;
+}
+
+export const USER_SENT_PROMPT_SCHEMA: UserInteractionTelemetryEvent = {
+  eventType: USER_SENT_PROMPT,
+  schema: {
+    '@timestamp': {
+      type: 'text',
+      _meta: {
+        description: 'The timestamp of the message.',
+      },
+    },
+    message: {
```

Comment:
Seeing how these are typed using the same `RootSchema<Message>`, does that mean some of this is duplicated in the event for feedback? Would it make sense to re-use a single definition?

Reply:
One uses `RootSchema<FeedbackMessage>`, the other uses `RootSchema<Message>`, so they are not the same.